### PR TITLE
CanPickup/canPickUp -> Equipable/canEquip

### PR DESCRIPTION
**Pull request by sylv256**
# Changes
- LivingEntity.canPickUp -> canEquip
- EntityPredicates.CanPickup -> Equipable

#1614

### CODE REVIEW

**Review by Juuxel — APPROVED** [review]

**Comment by sylv256** [comment]
According to people on the Fabricord server, modded entities and piglins can pick up items (like gold nuggets) without equipping them. I'm going to leave the PR open just in case I'm not entirely correct. However, I don't think canEquip is the right name.

**Comment by sylv256** [comment]
I think `canTake` would be better since it's easier to type than canPickUp and makes more sense than canEquip. I'll close this PR and make a new one.

**Comment by Juuxel** [comment]
`canPickUp` is only used for equipping checks. The method used for regular pick up checks is `canGather`, which is the one piglins, piglin brutes and villagers override to make gathering gold and food possible.

**Comment by sylv256** [comment]
Piglins and other entities don't actually use this method for picking up items. It's only for equipping items on entities such as armor stands and also equipping items on entities via a dispenser.

**Review by Virtuoel — APPROVED** [review]

File: mappings/net/minecraft/predicate/entity/EntityPredicates.mapping [hunk 1]
```diff
@@ -41,7 +41,7 @@ CLASS net/minecraft/class_1301 net/minecraft/predicate/entity/EntityPredicates
 		ARG 0 entity
 	METHOD method_5913 rides (Lnet/minecraft/class_1297;)Ljava/util/function/Predicate;
 		ARG 0 entity
-	CLASS class_1302 CanPickup
+	CLASS class_1302 CanEquip
```
Comment (liach) [inline mappings/net/minecraft/predicate/entity/EntityPredicates.mapping]: I'd suggest `Equipable` name instead

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/predicate/entity/EntityPredicates.mapping (hunk 1 above)
Comment (sylv256) [reply mappings/net/minecraft/predicate/entity/EntityPredicates.mapping]: Alright, I'll change the mappings real quick.

**Review by Mysterious-Dev — APPROVED** [review]

**Review by liach — APPROVED** [review]

**Pull request merged**